pkg/agent/controller/networkpolicy/l7engine/reconciler.go

```diff
@@ -0,0 +1,398 @@
+		return fmt.Errorf("failed to unregister handler for tenant %d with vlan %d: %v", vlanID, vlanID, output)
+	}
+
+	// Unregister the tenant.
+	scCmd = fmt.Sprintf("unregister-tenant %d", vlanID)
+	if output, err := r.suricataScFn(scCmd); err != nil {
+		return fmt.Errorf("failed to unregister tenant %d: %v", vlanID, output)
+	}
+
+	// Delete the tenant config file.
+	configPath := generateTenantConfigPath(vlanID)
+	if err := defaultFS.Remove(configPath); err != nil {
+		return fmt.Errorf("failed to delete tenant config file %s: %v", configPath, err)
+	}
+	return nil
+}
+
+func (r *Reconciler) initializeSuricata() {
+	// Create the config file /etc/suricata/antrea.yaml for Antrea which will be included in the default Suricata config file
+	// /etc/suricata/suricata.yaml.
+	suricataAntreaConfigData := fmt.Sprintf(`%%YAML 1.1
+---
+af-packet:
+  - interface: %[1]s
+    threads: auto
+    cluster-id: 80
+    cluster-type: cluster_flow
+    defrag: no
+    use-mmap: yes
+    tpacket-v2: yes
+    checksum-checks: no
+    bpf-filter: "ip or ip6"
+    copy-mode: ips
+    copy-iface: %[2]s
+  - interface:  %[2]s
+    threads: auto
+    cluster-id: 81
+    cluster-type: cluster_flow
+    defrag: no
+    use-mmap: yes
+    tpacket-v2: yes
+    checksum-checks: no
+    bpf-filter: "ip or ip6"
+    copy-mode: ips
+    copy-iface: %[1]s
+multi-detect:
+  enabled: yes
+  selector: vlan
+`, config.L7NetworkPolicyTargetPortName, config.L7NetworkPolicyReturnPortName) // TODO: expand loaders when there is not enough loaders
+	f, err := defaultFS.Create(antreaSuricataConfigPath)
+	if err != nil {
+		klog.ErrorS(err, "Failed to create Suricata config file", "FilePath", antreaSuricataConfigPath)
+		return
+	}
+	defer f.Close()
+	if _, err = f.WriteString(suricataAntreaConfigData); err != nil {
+		klog.ErrorS(err, "Failed to write Suricata config file", "FilePath", antreaSuricataConfigPath)
+		return
+	}
+
+	// Open the default Suricata config file /etc/suricata/suricata.yaml.
+	f, err = defaultFS.OpenFile(defaultSuricataConfigPath, os.O_APPEND|os.O_WRONLY, 0600)
+	if err != nil {
+		klog.ErrorS(err, "Failed to open default Suricata config file", "FilePath", defaultSuricataConfigPath)
+		return
+	}
+	defer f.Close()
+	// Include the config file /etc/suricata/antrea.yaml for Antrea in the default Suricata config file /etc/suricata/suricata.yaml.
+	if _, err = f.WriteString(fmt.Sprintf("include: %s\n", antreaSuricataConfigPath)); err != nil {
+		klog.ErrorS(err, "Failed to update default Suricata config file", "FilePath", defaultSuricataConfigPath)
+		return
+	}
+
+	r.startSuricataFn()
+
+	// Wait Suricata command socket file to be ready.
+	err = wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (bool, error) {
+		if _, err = defaultFS.Stat(suricataCommandSocket); err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		klog.ErrorS(err, "Failed to find Suricata command socket file")
+	}
```

Comment:
It would be confusing to see both "Failed to find Suricata command socket file" and "Started Suricata instance successfully" when it fails.

Reply:
Moved "Started Suricata instance successfully" to else branch.